### PR TITLE
Added Glob pattern matching for "Windows Services" plugin

### PR DESCRIPTION
**Pull request by sspaink**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

This pull request addresses the issue  #3673
Updated the "Windows Services" plugin to allow glob patterns in the input config for including services.
I based the changes on the Docker plugin which also allows glob pattern matching.

Tested the changes locally and uploaded some information:
```
[[inputs.win_services]]
  ## Names of the services to monitor. Leave empty to monitor all the available services on the host
  service_names = [
    "Win*",
  ]
```
image

### CODE REVIEW

File: plugins/inputs/win_services/README.md [hunk 1]
```diff
@@ -8,10 +8,11 @@ Monitoring some services may require running Telegraf with administrator privile
 
 ```toml
 [[inputs.win_services]]
-  ## Names of the services to monitor. Leave empty to monitor all the available services on the host
+  ## Names of the services to monitor. Leave empty to monitor all the available services on the host. Globs accepted.
```
Comment (ssoroka) [inline plugins/inputs/win_services/README.md:11]: I think globs are the technical name that isn't really used outside tech circles. Let's maybe refer to it as wildcards in the docs?

File: plugins/inputs/win_services/win_services.go [hunk 2]
```diff
@@ -117,7 +122,16 @@ func (m *WinServices) Gather(acc telegraf.Accumulator) error {
...
 
-	serviceNames, err := listServices(scmgr, m.ServiceNames)
+	// Create label filters if not already created
+	if !m.filtersCreated {
+		err := m.createServicesFilter()
+		if err != nil {
+			return err
+		}
+		m.filtersCreated = true
+	}
```
Comment (ssoroka) [inline plugins/inputs/win_services/win_services.go:132]: let's move this to an Init() function as it only has to happen once, and then you don't need the boolean guard around it.

File: plugins/inputs/win_services/win_services.go [hunk 3]
```diff
@@ -151,17 +165,30 @@ func (m *WinServices) Gather(acc telegraf.Accumulator) error {
...
-	if len(userServices) != 0 {
-		return userServices, nil
+func (m *WinServices) createServicesFilter() error {
+	filter, err := filter.NewIncludeExcludeFilter(m.ServiceNames, nil)
+	if err != nil {
+		return err
 	}
+	m.servicesFilter = filter
+	return nil
+}
```
Comment (ssoroka) [inline plugins/inputs/win_services/win_services.go:175]: can probably collapse this all into one line, might not need a function for it.

File: plugins/inputs/win_services/README.md (hunk 1 above)
Comment (sspaink) [reply plugins/inputs/win_services/README.md:11]: I agree, the word "wildcard" would probably be easier to understand. Although the example config that Telegraf produces does use the word "glob" multiple times, do you think we should make an issue to update it as well? 
Example: https://github.com/influxdata/telegraf/blob/7d3b7fc2f9d379aa21d09eb3e905e644e876be6f/etc/telegraf.conf#L2949

Comment (ssoroka) [reply plugins/inputs/win_services/README.md:11]: ok perhaps it's a larger discussion. feel free to ignore

File: plugins/inputs/win_services/win_services.go (hunk 2 above)
Comment (sspaink) [reply plugins/inputs/win_services/win_services.go]: Updated like suggested, Thanks!

File: plugins/inputs/win_services/README.md (hunk 1 above)
Comment (sspaink) [reply plugins/inputs/win_services/README.md:11]: ignoring for now then.

File: plugins/inputs/win_services/win_services.go (hunk 3 above)
Comment (sspaink) [reply plugins/inputs/win_services/win_services.go]: Updated like suggested, thanks!

**Review by ssoroka — APPROVED** [review]

**Pull request merged**